templates/base.html.twig

```diff
@@ -80,26 +80,48 @@
                                         <span class="caret"></span>
                                         <span class="sr-only">{{ 'menu.choose_language'|trans }}</span>
                                     </a>
-                                    <ul class="dropdown-menu locales" role="menu" aria-labelledby="locales">
-                                        {% for locale in locales() %}
-                                            <li {% if app.request.locale == locale.code %}aria-checked="true" class="dropdown-item active"{% else %}class="dropdown-item" aria-checked="false"{% endif %} role="menuitem">
-                                                <a href="{{ path(app.request.get('_route', 'blog_index'), app.request.get('_route_params', [])|merge({_locale: locale.code})) }}">{{ locale.name|capitalize }}
-                                                    <small>{{ locale.code[0:2] }}</small>
-                                                </a>
-                                            </li>
-                                        {% endfor %}
-                                    </ul>
                                 </li>
                             </ul>
                         </div>
                     </div>
                 </nav>
             </header>
+
+            <div id="locale-selector-modal" class="modal" tabindex="-1" aria-hidden="true">
+                <div class="modal-lg modal-dialog modal-dialog-centered">
+                    <div class="modal-content">
+                        <div class="modal-header">
+                            <h5 class="modal-title">
+                                {# display this title always in English to make the language selector easier to use #}
+                                Choose your language
+                            </h5>
+                            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                                <span aria-hidden="true">&times;</span>
+                            </button>
+                        </div>
+
+                        <div class="modal-body">
+                            <ul class="locales" role="menu" aria-labelledby="locales">
+                                {% for locale in locales() %}
+                                    {% set is_active = app.request.locale == locale.code %}
+                                    {% set is_rtl = locale.code in ['ar', 'fa', 'he'] %}
+                                    <li aria-checked="{{ is_active ? 'true' : 'false' }}" class="{{ is_active ? 'active' }} {{ is_rtl ? 'rtl' }}" role="menuitem" translate="no">
+                                        <a class="stretched-link" href="{{ path(app.request.get('_route', 'blog_index'), app.request.get('_route_params', [])|merge({_locale: locale.code})) }}">
+                                            <small>{{ locale.code[0:2] }}</small>
+                                            {{ locale.name|capitalize }}
```

Comment:
As this displays each locale name in its own language, we might need to add the appropriate `lang` attribute.

Looking at the europa.eu selector, they do have this `lang` attribute, and then also use `hreflang` on the link going to a different language.